src/vs/editor/browser/services/openerService.ts

```diff
@@ -51,12 +56,42 @@ export class OpenerService implements IOpenerService {
 
 	private _doOpen(resource: URI, options?: { openToSide?: boolean }): Promise<boolean> {
 
-		const { scheme, path, query, fragment } = resource;
+		const { scheme, authority, path, query, fragment } = resource;
 
-		if (equalsIgnoreCase(scheme, Schemas.http) || equalsIgnoreCase(scheme, Schemas.https) || equalsIgnoreCase(scheme, Schemas.mailto)) {
-			// open http or default mail application
+		if (equalsIgnoreCase(scheme, Schemas.mailto)) {
+			// open default mail application
 			return this.openExternal(resource);
+		}
 
+		if (equalsIgnoreCase(scheme, Schemas.http) || equalsIgnoreCase(scheme, Schemas.https)) {
+			const trustedDomains = this._configurationService.getValue<string[]>('http.trustedDomains');
+			const domainToOpen = `${scheme}://${authority}`;
+
+			if (isDomainTrusted(domainToOpen, trustedDomains)) {
+				return this.openExternal(resource);
+			} else {
+				return this._dialogService.confirm({
+					title: localize('openExternalLink', 'Open External Link'),
+					type: 'question',
+					message: localize('openExternalLinkAt', 'Do you want to leave VS Code to open the external website at') + ` ${resource.toString()}?`,
+					detail: resource.toString(),
+					primaryButton: localize('openLink', 'Open Link'),
+					secondaryButton: localize('cancel', 'Cancel'),
+					checkbox: {
+						label: localize('trustAllLinksOn', 'Trust all links on') + ` ${domainToOpen}`,
+						checked: false
+					}
+				}).then(({ confirmed, checkboxChecked }) => {
+					if (checkboxChecked) {
+						this._configurationService.updateValue('http.trustedDomains', [...trustedDomains, domainToOpen]);
```

Comment:
So, that extensions cannot mess with this, use `IStorageService`, with the global-scope.

Reply:
Done.